src/transformers/models/aria/modeling_aria.py

```diff
@@ -825,7 +833,10 @@ def forward(
         hidden_states = inputs_embeds
 
         # create position embeddings to be shared across the decoder layers
-        position_embeddings = self.rotary_emb(hidden_states, position_ids)
+        if cos is None or sin is None:
+            cos, sin = self.rotary_emb(hidden_states, position_ids)
+
+        position_embeddings = (cos, sin)
```

Comment:
happy to merge but let's just pass positionembeddings as a tuple rather than split. No need for doc in the modeling just update it in https://github.com/huggingface/transformers/blob/tp-cb/src/transformers/utils/args_doc.py#L208-L208